var/spack/repos/builtin/packages/py-tensorboard/package.py

```diff
@@ -3,38 +3,33 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
-import tempfile
-
 from spack.package import *
 
 
-class PyTensorboard(Package):
+class PyTensorboard(PythonPackage):
     """TensorBoard is a suite of web applications for
     inspecting and understanding your TensorFlow runs and
     graphs."""
 
     homepage = "https://github.com/tensorflow/tensorboard"
-    url      = "https://github.com/tensorflow/tensorboard/archive/2.2.0.tar.gz"
+    url = "https://files.pythonhosted.org/packages/py3/t/tensorboard/tensorboard-2.9.1-py3-none-any.whl"
+    list_url = "https://pypi.org/simple/tensorboard/"
```

Comment:
Does using wheel eliminate the `pypi` attribute?

Reply:
In the current implementation, yes. It turns out that it's actually quite difficult to programmatically parse the full wheel filename format. For `.tar.gz` and `.zip`, the string after "packages" is always "source". But for wheels, it could be "py3", "py2.py3", "cp37", etc. I need to make `pypi` smarter to handle these properly.